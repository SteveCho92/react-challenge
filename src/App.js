import React, { useState } from "react";
import "./index.css";

export default function App() {
  //functional components with states must use React Hooks useState to set initial state.
  //To store multiple values in a state, you can have it as a single object.
  //To extend more input fields on the form, you can add more properties to the single state object.
  const [form, setForm] = useState({
    name: '',
    age: '',
    addtionalField: 'something you can add!'
  });

  //updateForm is a general input change handler to update UI.
  //...form spread operator to maintain unchanged state
  //[e.target.name] must equal name of state property from useState
  const updateForm = e => {
    setForm({
      ...form,
      [e.target.name]: e.target.value
    });
  };

  return (
    <div className="App">
      <div>
        <h2 className="subtitle">React Challenge! Good Luck</h2>
      </div>
      <div className="content">
        {/* Display Data */}
        <div className="input-display">
          {/* You can access state in functional components directly without this.state */}
          <p data-testid="name-display">Name: {form.name}</p>
          <p data-testid="age-display">Age: {form.age}</p>
        </div>

        {/* Collect User Inputs */}
        <div className="inputs">
          {/* Input name */}
          <div className="field">
            <label className="label">Name: </label>
            <input
              className="input"
              name='name'
              type="text"
              placeholder="Enter your name, ex. Kevin"
              onChange={updateForm} 
              data-testid = "name-input" />
          </div>

          {/* Input age */}
          <div className="field">
            <label className="label">&nbsp;&nbsp;&nbsp;Age: </label>
            <input
              className="input"
              name='age'
              type="number"
              placeholder="Enter your age, ex. 44"
              onChange={updateForm} 
              min="0" 
              data-testid = "age-input" />
          </div>
        </div>
      </div>
    </div>
  );
}