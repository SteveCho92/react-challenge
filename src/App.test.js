import React from 'react';
import { render, fireEvent, cleanup } from '@testing-library/react';
import App from './App';
import TestRenderer from 'react-test-renderer';

afterEach(cleanup);

test('rendering App component', () => {
  const { getByText } = render(<App/>)
  expect(getByText(/React Challenge! Good Luck/i)).toBeInTheDocument();
});

//tests for changes made to component via snapshot
test('App component snapshot', () => {
  const AppSnapshot = TestRenderer.create(<App/>).toJSON();
  expect(AppSnapshot).toMatchSnapshot();
})

test("name input change and display", () => {
  const { getByTestId } = render(<App />);
  const inputName = getByTestId("name-input");

  const testName = "testing name";
  fireEvent.change(inputName, { target: { value: testName } });

  const labelName = getByTestId("name-display");

  expect(labelName.textContent).toBe("Name: " + testName);
});

test("age input change and display", () => {
  const { getByTestId } = render(<App />);
  const inputAge = getByTestId("age-input");

  const testAge = 10;
  fireEvent.change(inputAge, { target: { value: testAge } });

  const labelAge = getByTestId("age-display");
  expect(labelAge.textContent).toBe("Age: " + testAge);
});
