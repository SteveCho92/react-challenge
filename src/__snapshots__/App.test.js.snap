// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component snapshot 1`] = `
<div
  className="App"
>
  <div>
    <h2
      className="subtitle"
    >
      React Challenge! Good Luck
    </h2>
  </div>
  <div
    className="content"
  >
    <div
      className="input-display"
    >
      <p
        data-testid="name-display"
      >
        Name: 
        
      </p>
      <p
        data-testid="age-display"
      >
        Age: 
        
      </p>
    </div>
    <div
      className="inputs"
    >
      <div
        className="field"
      >
        <label
          className="label"
        >
          Name: 
        </label>
        <input
          className="input"
          data-testid="name-input"
          name="name"
          onChange={[Function]}
          placeholder="Enter your name, ex. Kevin"
          type="text"
        />
      </div>
      <div
        className="field"
      >
        <label
          className="label"
        >
             Age: 
        </label>
        <input
          className="input"
          data-testid="age-input"
          min="0"
          name="age"
          onChange={[Function]}
          placeholder="Enter your age, ex. 44"
          type="number"
        />
      </div>
    </div>
  </div>
</div>
`;
